map/src/login/DownloadBackupDialog.jsx

```diff
@@ -204,6 +217,17 @@ export default function DownloadBackupDialog({ openDownloadBackupDialog, setOpen
         }
     }
 
+    function selectAllCategory(type, res, groups) {
```

Comment:
As for me, it's not good idea to modify array/object which is parameter of the function.

This is OK only when it's absolutely clear by code (for example, rename function to selectAllCategoryAndUpdateRes()

In your case, it's better to make new array inside function, push res to it, then push other elements, return it and use result by parent as new array.